\begin{abstract}
de Bruijn graphs are nowadays an inseparable part of Next Generation Sequence analyses. With the fast growth in the amount of sequencing reads and variety of different genomes and transcriptomes the community is shifting toward using graphs to represent collections of references and map reads. 
In this document, we present two succinct representations for two different variations of a de Bruijn graph, namely colored de Bruijn graph and compacted de Bruijn graph. For the former, we designed and developed rainbowfish, a succinct data structure to represent colors for an efficient \dbg representation and also theoretically proved why we call it succinct. This structure is useful in genome variant detection and genotyping. For the later, we developed a tool named pufferfish for indexing a compacted de Bruijn graph in two different schemes of dense and sparse. While being close to linear indexing methodologies regarding memory, pufferfish shows to have a similar \kmer lookup speed as other memory-consuming de Bruijn graph indexing schemes. The balance that pufferfish offers between memory and speed makes it suitable for large-scale indexing such as for collections of RNA-seq data or in metagenomic analysis.
\end{abstract} 